docs/REVIEWING.md

```diff
@@ -36,6 +36,40 @@ following items should also be checked:
 - [ ] Does it comply with [component spec](specs/component.md)?
 - [ ] Does it comply with the [instrumentation spec](specs/instrumentation.md)?
 
+
+### Sink review checklist
+
+This checklist is specific for Vector's sink code.
+
+## Logic
+
+- [ ] Does it work? Do you understand what it is supposed to be doing?
+- [ ] Does the retry logic make sense?
+- [ ] Are the tests testing that the sink is emitting the correct metrics?
+- [ ] Are there integration tests?
+
+## Code structure
+
+- [ ] Is it using the sink prelude?
+- [ ] Is it gated by sensible feature flags?
+- [ ] Is the code modularized into `mod.rs`, `config.rs`, `sink.rs`,  `request_builder.rs`, `service.rs`
+- [ ] Does the code follow our [style guidelines].
+
+## Documentation
+
+- [ ] Look at the doc preview on Netlify. Does it look good? 
+- [ ] Is there a `cue` file linking to `base`?
```

Comment:
Does this imply/cover the auto generated docs?

Reply:
I would say so. CI will catch it if it they don't run `make generate-component-docs`, and looking at the doc preview should ensure that the docs are sensibly written.